BUGFIX: Fusion avoid error parser cache to crash if cache is broken

mhsdesign opened the pull request:
Resolves #4595

It seems a cache's `has` operation doesnt work that reliable thats why we will be using `get` instead and check against false.  

<!-- 
    Thanks for your contribution, we appreciate it!

    The first section should explain briefly what is changed. 
    Some examples are always nice to showcase the use. 
    The content will be used in the change-logs and addresses 
    developers working with Neos.

    If there are issues regarding the topic of your PR link 
    them here as `related:` or `resolved:`
-->

**Upgrade instructions**

<!-- 
    Add upgrade instructions for breaking changes. 
    Explain who is affected, what has to be adjusted  
-->

**Review instructions**

<!-- 
    If your change is not explained fully by the first section you can 
    add more details here to help the reviewers understand the change.
    We have to understand what you did, why you did it and how we can 
    verify it works correctly and does no harm.
-->

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

mhsdesign committed:
BUGFIX: Fusion avoid error parser cache to crash if cache is broken

Resolves #4595

Sebobo reviewed (approved):
LGTM

mhsdesign commented on Neos.Fusion/Classes/Core/Cache/ParserCache.php:
```diff
@@ -59,7 +67,18 @@ public function cacheForFusionFile(?string $contextPathAndFilename, \Closure $ge
-        return $this->cacheForIdentifier($identifier, $generateValueToCache);
+        $value = $this->cacheForIdentifier($identifier, $generateValueToCache);
+        if (!$value instanceof FusionFile) {
+            $this->logger->error(sprintf(
```
maybe we should even use the throwable error storage to empathise on the error?

kitsunet replied:
I wouldn't even categorize this as error. It's a cache, things can happen, it should be able to deal with this.

mhsdesign replied:
okay now that you say this, i agree.

> Is why I almost never use "has" in caches, it is never atomic, you cannot know what happens between has and get.

will update this accordingly. Is there anything we have to fix in flow then for 9.0? Like remove has :D (No just joking as _false_ could be an acceptable value...

mhsdesign replied:
removed the logging ;)

mhsdesign committed:
TASK: Alternate fix for fusion partials cache #4595

> This should not be considered an error. It's a cache, things can happen, it should be able to deal with this.

That's why get will be used directly instead of has.

In the rare edge-case of a fusion dsl returning `false` we cannot cache it anymore.
This is an acceptable compromise.
